docs: add v0.8.0 highlights (#972)
GreptimeTeam/docs: docs/release-notes/release-0-8-0.md

# v0.8.0

Release date: May 18, 2024

## 👍 Highlights

> - Flow Engine: which brings continuous aggregation capabilities.

Reviewer: Can we add some doc links to these highlighted features if they exist?

> - Flow Engine: which brings continuous aggregation capabilities.

Reviewer 2: ```suggestion
- [Flow Engine](/user-guide/continuous-aggregation/overview): which brings continuous aggregation capabilities.
```

Reviewer 2: @tisonkun I'm not sure if these links work

Reviewer 3: This should work.

I suggest @WenyXu to create the PR based on the upstream branches so that Cloudflare preview can work.

Reviewer 3: However, I may suggest to use `https://docs.greptime.com/user-guide/...` to keep links to the latest stable version.

> - Column Type Modification: allows users to effortlessly alter the data type of columns within a table without the hassle of rebuilding the table or manually migrating data.

Reviewer 2: ```suggestion
- [Column Type Modification](/reference/sql/alter#modify-column-type): allows users to effortlessly alter the data type of columns within a table without the hassle of rebuilding the table or manually migrating data.
```

> - Cluster Management Information Table: allows querying for information about the cluster. This functionality aids administrators in monitoring and managing the health status of the database cluster, facilitating prompt issue detection and resolution.

Reviewer 2: ```suggestion
- [Cluster Management Information Table](/reference/sql/information-schema/overview): allows querying for information about the cluster. This functionality aids administrators in monitoring and managing the health status of the database cluster, facilitating prompt issue detection and resolution.
```

> - Append-only Tables: Users can now create tables in Append-only mode by setting the append mode during table creation.

Reviewer 2: ```suggestion
- [Append-only Tables](/reference/sql/create#table-options): You can now create tables in Append-only mode by setting the append mode during table creation.
```

> - DROP DATABASE: enables swift deletion of all tables and resources under a database instance.

Reviewer 2: ```suggestion
- [DROP DATABASE](/reference/sql/drop.md#drop-database): enables swift deletion of all tables and resources under a database instance.
```

> - New Table Partitioning Methods and Syntax: considering the future need for frequent partition changes such as automatic partitioning or repartitioning, we're developing a new partitioning syntax.

Reviewer 2: ```suggestion
```

> - `EXPLAIN ANALYZE <QUERY>`: analyze and optimize query statements swiftly in distributed mode.

Reviewer 2: ```suggestion
- [`EXPLAIN ANALYZE <QUERY>`](/reference/sql/explain): analyze and optimize query statements swiftly in distributed mode.
```
